tests/test_worker.py

```diff
@@ -101,3 +100,18 @@ def test_worker_list_resources(hq_env: HqEnv):
     assert table[0][3] == "Resources"
     assert table[1][3] == "1x10 cpus"
     assert table[2][3] == "4x5 cpus"
+
+
+def test_idle_timeout(hq_env: HqEnv):
+    hq_env.start_server(args=["--idle-timeout", "1s"])
+    w = hq_env.start_worker(args=["--heartbeat", "500ms"])
+    time.sleep(0.5)
+    hq_env.command(["submit", "--", "sleep", "1"])
+    time.sleep(1.0)
```

Comment:
`wait_for_job_state`?

Reply:
I am testing here that worker is not turned off too soon. So putting here an wait_for_job state could be placed here, but still some sleep is required

Reply:
Yeah just the `time.sleep(1.0)` should be replaced with `wait_for_job_state(hq_env, 1, "FINISHED")`.